CONTRIBUTING.md

```diff
@@ -82,13 +82,20 @@ changes.
 
 ### Developing and Running SQLFluff Locally
 
+#### Requirements
+
 The simplest way to set up a development environment is to use `tox`.
-First ensure that you have tox installed (windows users may have to replace `python3` with `py`):
+First ensure that you have tox installed:
 ```shell
-python3 -m pip install -U tox
+python3.8 -m pip install -U tox
 ```
+**IMPORTANT:** `tox` must be installed using Python 3.8 or newer. Otherwise `mypy` checks may fail. Depending on your setup and personal preference, replace `python3.8` as necessary, but make sure that the python version is new enough.
+
+<!-- Hopefully when tox 4 is released it will be possible to set `basepython = python3.10, python3.9, python3.8` to enforce a working version for running mypy checks. For now it's only possible to set it to a single version which is too restrictive. On the other hand for now there is a requirement for contributors to install tox with a new enough Python version. -->
 
-A virtual environment can then be created and activated by running:
+#### Creating a virtual environment
+
+A virtual environment can then be created and activated by running (check the [requirements](#requirements) before running this):
```

Comment:
Not sure this is needed and reads a little confusing. I think the `Then` covers it.

```suggestion
A virtual environment can then be created and activated by running:
```

Reply:
I wanted to be extra sure. If someone just jumps to this section and happens to have tox installed previously, they might miss the requirement. Are you ok with keeping the pointer?

Reply:
Yup. OK keeping them all as you prefer.